File: python/robot-name/robot_name.py
from random import sample, seed
from string import ascii_uppercase as uppercase, digits

class Robot(object):
    def __init__(self):
        self._generate_name()

    def reset(self):
        seed(self.name)
        self._generate_name()

    def _generate_name(self):
        letters = ''.join(sample(uppercase, 2))
        numbers = ''.join(sample(digits, 3))
        self.name = letters + numbers

File: javascript/robot-name/robot-name.js
let names = new Set();
let letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
let numbers = '0123456789';

function makeName() {
    let name = randomChars(letters, 2) + randomChars(numbers, 3);
    if (names.has(name)) {
        return makeName();
    }
    names.add(name);
    return name;
}

function randomChars(str, len) {
    return Array(len)
        .fill(null)
        .map(_ => str.charAt(Math.floor(Math.random() * str.length)))
        .join('');
}

class RobotName {

    constructor() {
        this.reset();
    }

    reset() {
        this.name = makeName();
    }

}

module.exports = RobotName;